.github/workflows/build.yml

```diff
@@ -7,54 +8,47 @@ on:
     branches:
     - main
 
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
 jobs:
   build:
-    name: Build ${{ matrix.os }} python ${{ matrix.python }}
+    name: Build - ${{ matrix.os }} - python ${{ matrix.python }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [openstack22]
-        python: ["3.12"]
-    timeout-minutes: 500
+        os:
+          - ubuntu-latest
+          # `ubuntu-22.04-arm` is considered more stable than `ubuntu-24.04-arm`:
+          # <https://github.com/orgs/community/discussions/148648#discussioncomment-12099554>.
+          - ubuntu-22.04-arm
+          - macOS-latest
+        python:
+          - '3.12'
+    timeout-minutes: 180
     steps:
-      - name: add llvm
-        run: |
-            if [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
-              sudo apt-get update
-              sudo apt-get install -y git gcc g++ #python3 python3-dev python3-pip
-              #sudo python3 -m pip install --upgrade lit
-              #- run: python3 -m pip install --user numpy
-            fi
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           submodules: recursive
-      
-      - run: sudo rm -rf ~/.cache/bazel ~/.cache/bazelisk
-      
-      - uses: bazel-contrib/setup-bazel@0.8.5
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: bazel-contrib/setup-bazel@0.13.0
         name: Set up Bazel
         with:
           # Avoid downloading Bazel every time.
           bazelisk-cache: true
           # Store build cache per workflow.
-          # disk-cache: ${{ github.workflow }}-${{ hashFiles('.github/workflows/build.yml') }}
+          disk-cache: ${{ github.workflow }}-${{ matrix.os }}-${{ matrix.python }}-${{ hashFiles('.github/workflows/build.yml') }}
           # Share repository cache between workflows.
           repository-cache: true
           bazelisk-version: 1.x
-
-      - name: test
-        run: |
-          HERMETIC_PYTHON_VERSION=${{ matrix.python }} bazel test --define=xnn_enable_avxvnniint8=false --define=xnn_enable_avx512fp16=false --test_output=errors ...
-          
-      - name: Build Wheel
+      - name: Build enzymexlamlir-opt
         run: |
-          sudo rm bazel-bin/*.whl || echo
-          HERMETIC_PYTHON_VERSION=${{ matrix.python }} bazel build --define=xnn_enable_avxvnniint8=false --define=xnn_enable_avx512fp16=false :wheel
-          cp bazel-bin/*.whl .
-
-      - name: Upload Build
-        uses: actions/upload-artifact@v4
-        with:
-          name: Wheel-${{ matrix.os }}
-          path: "*.whl"
+          bazel build --color=yes //:enzymexlamlir-opt --test_output=errors
```

Comment:
we need to retain the wheel building capabilities as well, if that can be added (in particular the actual python pip package, which while julia isn't using, shouldn't be dropped)

Reply:
`bazel build :wheel`? Something else? Can that be a separate step after the first build?